jobs/shield-agent/templates/bin/post-start.erb

```diff
@@ -9,21 +9,33 @@ done
 <% if_p("shield.agent.autoprovision") do |server| %>
 <% if_p("shield.target.name") do |target_name|%>
 
-export SHIELD_API=<%= server %>
+<%#
+  The SHIELD_API_TOKEN is set here so that the post-start script will fail to render
+  when provisioning_key is not specified, but the script is about to try to create targets
+  or jobs
+%>
+export SHIELD_API_TOKEN=<%= p("shield.provisioning_key") %>
 
-TARGET=$(shield --raw show target <%= target_name %> | jq -r '.uuid // empty')
+
+<% if p("shield.skip_ssl_verify") %>
+export SHIELD_SKIP_SSL_VERIFY=true
+<% end %>
+
+shield -c /tmp/.shield_agent_config create backend default https://<%= p("shield.daemon.domain") %>:<%= p("shield.daemon.port") %>
```

Comment:
Can you put a note in as to why the config file is created via `shield`, and put in `/tmp/` vs. `$HOME`?  That way we (I) don't try to fix that in the future...